generator/uscore/uscore_generator.rb

```diff
@@ -131,10 +131,21 @@ def create_authorization_test(sequence)
         first_search = find_first_search(sequence)
         return if first_search.nil?
 
+        search_parameters = first_search[:names]
+        search_params = if search_parameters == ['patient'] || sequence[:delayed_sequence] || search_param_constants(search_parameters, sequence)
```

Comment:
I think all of the logic for what search params should be used belongs in `get_search_param_hash`. This method should be able to just call `get_search_params` and `get_search_param_hash`.

Reply:
The problem with this is that sometimes get_search_params needs the value from resolve_element_from_path and here it needs it from the fixed code. I'd have to add a thing just for the case of the authorization test. I think I can get around this by just putting the authorization test after the search tests which I actually like. I'll try that out and put it in this branch.

Reply:
This logic only depends on `search_parameters` and `sequence`, which are the arguments which `get_search_param_hash` already takes, so it should be possible to move the logic there.

Reply:
yep, much cleaner now.